homeassistant/components/synology_dsm/__init__.py

```diff
@@ -604,51 +598,23 @@ async def async_update(self, now: timedelta | None = None) -> None:
 class SynologyDSMBaseEntity(CoordinatorEntity):
     """Representation of a Synology NAS entry."""
 
+    entity_description: SynologyDSMEntityDescription
+    unique_id: str
+
     def __init__(
         self,
         api: SynoApi,
-        entity_type: str,
-        entity_info: EntityInfo,
         coordinator: DataUpdateCoordinator[dict[str, dict[str, Any]]],
+        description: SynologyDSMEntityDescription,
     ) -> None:
         """Initialize the Synology DSM entity."""
         super().__init__(coordinator)
+        self.entity_description = description
 
         self._api = api
-        self._api_key = entity_type.split(":")[0]
-        self.entity_type = entity_type.split(":")[-1]
-        self._name = f"{api.network.hostname} {entity_info[ATTR_NAME]}"
-        self._class = entity_info[ATTR_DEVICE_CLASS]
-        self._enable_default = entity_info[ENTITY_ENABLE]
-        self._icon = entity_info[ATTR_ICON]
-        self._unit = entity_info[ATTR_UNIT_OF_MEASUREMENT]
-        self._unique_id = f"{self._api.information.serial}_{entity_type}"
-        self._attr_state_class = entity_info[ATTR_STATE_CLASS]
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return self._unique_id
-
-    @property
-    def name(self) -> str:
-        """Return the name."""
-        return self._name
-
-    @property
-    def icon(self) -> str | None:
-        """Return the icon."""
-        return self._icon
-
-    @property
-    def device_class(self) -> str | None:
-        """Return the class of this device."""
-        return self._class
-
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the state attributes."""
-        return {ATTR_ATTRIBUTION: ATTRIBUTION}
+        self._attr_name = f"{api.network.hostname} {description.name}"
+        self._attr_unique_id: str = f"{api.information.serial}_{description.key}"
```

Comment:
```suggestion
        self._attr_unique_id: str = f"{api.information.serial}_{description.api_key}:{description.key}"
```

to match previous generated unique id's

Reply:
Good catch 👍🏻